r2/r2/lib/solrsearch.py

```diff
@@ -0,0 +1,564 @@
+# The contents of this file are subject to the Common Public Attribution
+# License Version 1.0. (the "License"); you may not use this file except in
+# compliance with the License. You may obtain a copy of the License at
+# http://code.reddit.com/LICENSE. The License is based on the Mozilla Public
+# License Version 1.1, but Sections 14 and 15 have been added to cover use of
+# software over a computer network and provide for limited attribution for the
+# Original Developer. In addition, Exhibit A has been modified to be consistent
+# with Exhibit B.
+#
+# Software distributed under the License is distributed on an "AS IS" basis,
+# WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+# the specific language governing rights and limitations under the License.
+#
+# The Original Code is reddit.
+#
+# The Original Developer is the Initial Developer.  The Initial Developer of
+# the Original Code is reddit Inc.
+#
+# All portions of the code written by reddit are Copyright (c) 2006-2015 reddit
+# Inc. All Rights Reserved.
+###############################################################################
+
+from datetime import datetime, timedelta
+import functools
+import httplib
+import json
+
+from lxml import etree
+from r2.lib import amqp, filters
+from r2.lib.db.sorts import epoch_seconds
+import r2.lib.utils as r2utils
+from r2.models import (Account, Subreddit, All, DefaultSR,
+                       MultiReddit, DomainSR, Friends, ModContribSR,
+                       FakeSubreddit)
+from pylons import g, c
+
+from r2.lib.cloudsearch import *
```

Comment:
Please don't use `import *`; it can cause subtle import bugs (such as shadowing the `datetime` class with the `datetime` module).